ci/compile_and_run.sh

```diff
@@ -13,10 +14,10 @@ ARGS=(
   -D DEAL_II_DIR=${DEAL_II_DIR}
   -D MFMG_ENABLE_AMGX=ON
   -D AMGX_DIR=${AMGX_DIR}
-  -D CMAKE_CXX_FLAGS="-Wall -Wpedantic -Wextra -Wshadow"
+  -D CMAKE_CXX_FLAGS="-Wall -Wpedantic -Wextra -Wshadow -Werror"
```

Comment:
Is that really a good idea?

Reply:
What are your objections?

Reply:
Won't even run the tests if you have a warning that you might have missed cause you use a different compiler

Reply:
For me, `CI` should check for all the requirements we want a pull request/the code to fulfill in the end and not so much as a development tool when creating a pull request.

In `deal.II`, we just have multiple checks were one would one would not have `-Werror` and run tests while another one would check, e.g. for indentation (not as a test) and for compiler warnings.

I would also be in favor to run `clang-tidy` if we want to enforce its checks in the end.

Reply:
What do you guys want to do here? I don't have a strong opinion either way